ros2bag/ros2bag/verb/record.py

```diff
@@ -52,6 +51,13 @@ def add_arguments(self, parser, cli_name):  # noqa: D102
             help='time in ms to wait between querying available topics for recording. It has no '
              'effect if --no-discovery is enabled.'
         )
+        parser.add_argument(
+            '-b', '--bag-size', type=int, default=sys.maxsize,
```

Comment:
There are multiple defaults values through the code: `sys.maxsize` there, then `UINT64_MAX`. Let's use '0' as the default and make this mean "no splitting".